src/wp-admin/includes/file.php

```diff
@@ -1126,7 +1126,11 @@ function download_url( $url, $timeout = 300, $signature_verification = false ) {
 		return new WP_Error( 'http_no_url', __( 'Invalid URL Provided.' ) );
 	}
 
-	$url_filename = basename( parse_url( $url, PHP_URL_PATH ) );
+	$url_path     = parse_url( $url, PHP_URL_PATH );
+	$url_filename = '';
+	if ( is_string( $url_path ) ) {
```

Comment:
I'd suggest checking that the `'' !== $url_path`. Why?

- No need to run through `basename` with an empty string
- And it'll use the same pattern as we've done elsewhere